src/libcore/option.rs

```diff
@@ -981,6 +981,102 @@ impl<T> From<T> for Option<T> {
     }
 }
 
+#[stable(feature = "rust1", since = "1.0.0")]
```

Comment:
All methods below essentially have the same code, maybe dedup them with a macro?